src/main/software/amazon/event/ruler/ComparableNumber.java

```diff
@@ -39,7 +39,29 @@ static String generate(final double f) {
             throw new IllegalArgumentException("Value must be between " + -Constants.FIVE_BILLION +
                     " and " + Constants.FIVE_BILLION + ", inclusive");
         }
-        return String.format("%014x", (long) (TEN_E_SIX * (Constants.FIVE_BILLION + f))).toUpperCase(Locale.ROOT);
+
+        return toHexStringSkippingFirstByte((long) (TEN_E_SIX * (Constants.FIVE_BILLION + f)));
+    }
+
+    private static final String HEXES = "0123456789ABCDEF";
+
+    private static byte[] longToByteBuffer(long value) {
+        ByteBuffer buffer = ByteBuffer.allocate(Long.BYTES);
+        buffer.putLong(value);
+        return buffer.array();
     }
+
+    static String toHexStringSkippingFirstByte(Long value) {
+        byte[] raw = longToByteBuffer(value);
+        char[] chars = new char[14];
+        for (int i = 1; i < raw.length; i++) {
+            byte b = raw[i];
+            int pos = (i - 1) * 2;
+            chars[pos] = HEXES.charAt((b & 0xF0) >> 4);
+            chars[pos + 1] = HEXES.charAt((b & 0x0F));
+        }
```

Comment:
not a blocker but would  prefer to have documentation around these two lines. Not all engineers get what's happening.